### PR TITLE
Login page fixes

### PR DESCRIPTION
**Pull request by alistair23**
This series provides a lot of fixes for Windows 10 and Windows Phone 10 UWP platforms.

This sets the toolbar options to have images (otherwise they look blank on the default view). See https://github.com/bitwarden/mobile/pull/188 for what the difference is.

This also fixes up the layout to correctly separate the login boxes (they used to be touching) and ensures that the reset password link can be seen by the user.

It fixes up some missing images.

Finally it allows the back button to appear when running on Windows 10. Previously it would not appear and there would be no obvious way to progress backwards. This allows us to remove the cancel button.

I'm getting the hang of this now, so if this is merged I can keep working on the other pages.

This is what it looks like now:
Phone
windowsphone-uwp-login
Desktop
windows10-uwp-login

### CODE REVIEW

File: src/App/Utilities/Extentions.cs [hunk 1]
```diff
@@ -55,7 +55,14 @@ public static void FocusWithDelay(this View view, int delay = 1000, bool forceDe
...
         public static async Task PushForDeviceAsync(this INavigation navigation, Page page)
         {
-            await navigation.PushModalAsync(new ExtendedNavigationPage(page), true);
+            if (Device.RuntimePlatform == Device.Windows)
+            {
+                await navigation.PushAsync(page, true);
+            }
+            else
+            {
+                await navigation.PushModalAsync(new ExtendedNavigationPage(page), true);
```
Comment (kspearrin) [inline src/App/Utilities/Extentions.cs]: Does UWP  not work with modals or something? I thought people were suppose to use the back button, like Android?

File: src/App/Utilities/Helpers.cs [hunk 2]
```diff
@@ -73,6 +73,17 @@ public static string GetEmptyTableSectionTitle()
 
             return " ";
         }
+
+        public static string LoginImage()
```
Comment (kspearrin) [inline src/App/Utilities/Helpers.cs]: I'd rather not have a helper for every type of image. Just create one method and pass the icon image string to it via parameter.

File: src/App/Pages/LoginPage.cs [hunk 3]
```diff
@@ -112,14 +112,20 @@ private void Init()
...
-            var loginToolbarItem = new ToolbarItem(AppResources.LogIn, null, async () =>
+            var loginToolbarItem = new ToolbarItem(AppResources.LogIn, Helpers.LoginImage(), async () =>
             {
                 await LogIn();
             }, ToolbarItemOrder.Default, 0);
 
             ToolbarItems.Add(loginToolbarItem);
             Title = AppResources.Bitwarden;
-            Content = scrollView;
+            if (Device.RuntimePlatform == Device.Windows)
```
Comment (kspearrin) [inline src/App/Pages/LoginPage.cs]: Why was this needed?

File: src/App/Pages/LoginPage.cs [hunk 4]
```diff
@@ -67,7 +67,7 @@ private void Init()
 
             var table = new ExtendedTableView
             {
-                Intent = TableIntent.Settings,
+                Intent = TableIntent.Form,
```
Comment (kspearrin) [inline src/App/Pages/LoginPage.cs]: Not sure how this will affect others.

Comment (alistair23) [reply src/App/Pages/LoginPage.cs]: This isn't required, it just seemed to make more sense then settings.

File: src/App/Pages/LoginPage.cs (hunk 3 above)
Comment (alistair23) [reply src/App/Pages/LoginPage.cs]: The scroll ends up being really weird without this. Instead of everything fitting in a single page (which it could) it gets expanded to scroll. The register page is even worse. Just run the UWP app and you will see the issue

File: src/App/Utilities/Extentions.cs (hunk 1 above)
Comment (alistair23) [reply src/App/Utilities/Extentions.cs]: They do use the back button. It comes up in the desktop version when there is something to go back to (the settings app does this). It only comes up with PushAsync though

File: src/App/Utilities/Helpers.cs (hunk 2 above)
Comment (alistair23) [reply src/App/Utilities/Helpers.cs]: I'm not completely sure what you mean

File: src/App/Pages/LoginPage.cs (hunk 4 above)
Comment (kspearrin) [reply src/App/Pages/LoginPage.cs]: This should be reverted.

File: src/App/Pages/LoginPage.cs (hunk 3 above)
Comment (kspearrin) [reply src/App/Pages/LoginPage.cs]: We likely need to do some work on the custom renderers for TableView like we have for the other platforms.

File: src/App/Utilities/Extentions.cs (hunk 1 above)
Comment (kspearrin) [reply src/App/Utilities/Extentions.cs]: I think we need a close button then like iOS. There are many pages that use modal that will be affected.

File: src/App/Pages/LoginPage.cs (hunk 4 above)
Comment (alistair23) [reply src/App/Pages/LoginPage.cs]: Ok, I can remove the patch

File: src/App/Pages/LoginPage.cs (hunk 3 above)
Comment (alistair23) [reply src/App/Pages/LoginPage.cs]: Yeah, I think that is the problem. I looked at the vault last night and that display is all messed up.

Where is a good place to start?

File: src/App/Utilities/Extentions.cs (hunk 1 above)
Comment (alistair23) [reply src/App/Utilities/Extentions.cs]: Do you mean for other pages that pop up or for this page?

**Comment by alistair23** [comment]
Ok, I dropped it down to just two patches, that aren't controversial.

I have some work in progress on the other issues and setting the custom renderer. It's here, but I don't think it's fully working yet: https://github.com/alistair23/mobile/tree/alistair/uwp-fixes

File: src/App/Utilities/Extentions.cs (hunk 1 above)
Comment (alistair23) [reply src/App/Utilities/Extentions.cs]: Looking at the Xamarin forms documentation this shouldn't be a modal push. Modals are for things that can't be navigated away from (hence the no back button). See here: https://developer.xamarin.com/guides/xamarin-forms/application-fundamentals/navigation/modal/

Would it be better to swap to only using modal sometimes? Either add an extra argument to specify non-modal or create a new function?
